soc/arm/nxp_imx/rt/CMakeLists.txt

```diff
@@ -11,7 +11,10 @@ zephyr_linker_sources_ifdef(CONFIG_NXP_IMX_RT_BOOT_HEADER
 
 # Enable custom OCRAM noncacheable region
 zephyr_linker_sources_ifdef(CONFIG_OCRAM_NOCACHE SECTIONS sections.ld)
-zephyr_sources_ifdef(CONFIG_OCRAM_NOCACHE mpu_regions.c)
+# MPU regions definition is required if openamp support is enabled, or for nocache region
+if(CONFIG_OCRAM_NOCACHE OR CONFIG_OPENAMP)
+zephyr_sources(mpu_regions.c)
```

Comment:
I updated the PR to use that. Note- the MPU region attribute for the shared memory needs to be `REGION_IO_ATTR` instead of `REGION_RAM_NOCACHE_ATTR`, so I updated the DT binding for `zephyr,memory-region` to allow for "IO" to be one of the possible `zephyr,memory-region-mpu` settings.